### PR TITLE
`scroll-reth` required features

### LINKED ISSUE #110: Add `required-features` field to `scroll-reth` `Cargo.toml`
### Describe the feature

We should add `required-features= ["scroll"]` to the `scroll-reth` binary `Cargo.toml` This will prevent the binary from being built without this feature flag which is a hard requirement. 

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by frisitano**
This PR introduces `required-features = ["scroll"]` on the `scroll-reth` binary. This is because the BMPT requires the `scroll` feature flag to operate correctly. 

closes: #110

### CODE REVIEW

File: crates/scroll/bin/scroll-reth/Cargo.toml [hunk 1]
```diff
@@ -38,4 +38,5 @@ optimism = [
 
 [[bin]]
 name = "scroll-reth"
-path = "src/main.rs"
\ No newline at end of file
+path = "src/main.rs"
+required-features = ["scroll"]
```
Comment (greged93) [inline crates/scroll/bin/scroll-reth/Cargo.toml:43]: nit: line break

**Review by greged93 — COMMENTED** [review]
just a nit on the line break

File: crates/scroll/bin/scroll-reth/Cargo.toml (hunk 1 above)
Comment (frisitano) [reply crates/scroll/bin/scroll-reth/Cargo.toml:43]: This was added intentionally.

Comment (greged93) [reply crates/scroll/bin/scroll-reth/Cargo.toml:43]: you mean you meant to remove the line break?

Comment (frisitano) [reply crates/scroll/bin/scroll-reth/Cargo.toml:43]: Oh sorry I thought you meant the introduction of the new line at eof. What do you mean about the line break?

Comment (greged93) [reply crates/scroll/bin/scroll-reth/Cargo.toml:43]: I mean you added a new line at the end of the file, we can merge as is for me, doesn't matter much

**Pull request merged**